diff --git a/dbus.te b/dbus.te
new file mode 100644
index 0000000..f26105a
--- /dev/null
+++ b/dbus.te
@@ -0,0 +1,4 @@
+type dbus_daemon, domain;
+type dbus_daemon_exec, exec_type, file_type;
+
+init_daemon_domain(dbus_daemon)
diff --git a/file.te b/file.te
index 24ca895..7eef0e8 100644
--- a/file.te
+++ b/file.te
@@ -138,6 +138,7 @@ type bluetooth_efs_file, file_type;
 # Socket types
 type adbd_socket, file_type;
 type bluetooth_socket, file_type;
+type dbus_daemon_socket, file_type;
 type dnsproxyd_socket, file_type, mlstrustedobject;
 type dumpstate_socket, file_type;
 type fwmarkd_socket, file_type, mlstrustedobject;
diff --git a/file_contexts b/file_contexts
index 62802e0..5ab5b41 100644
--- a/file_contexts
+++ b/file_contexts
@@ -77,6 +77,7 @@
 /dev/snd(/.*)?		u:object_r:audio_device:s0
 /dev/socket(/.*)?	u:object_r:socket_device:s0
 /dev/socket/adbd	u:object_r:adbd_socket:s0
+/dev/socket/dbus           u:object_r:dbus_daemon_socket:s0
 /dev/socket/dnsproxyd	u:object_r:dnsproxyd_socket:s0
 /dev/socket/dumpstate	u:object_r:dumpstate_socket:s0
 /dev/socket/fwmarkd	u:object_r:fwmarkd_socket:s0
@@ -135,6 +136,7 @@
 /system/bin/vold	u:object_r:vold_exec:s0
 /system/bin/netd	u:object_r:netd_exec:s0
 /system/bin/rild	u:object_r:rild_exec:s0
+/system/bin/dbus-daemon    u:object_r:dbus_daemon_exec:s0
 /system/bin/mediaserver	u:object_r:mediaserver_exec:s0
 /system/bin/mdnsd	u:object_r:mdnsd_exec:s0
 /system/bin/installd	u:object_r:installd_exec:s0
diff --git a/radio.te b/radio.te
index 7d90706..3f54cc5 100644
--- a/radio.te
+++ b/radio.te
@@ -11,6 +11,9 @@ unix_socket_connect(radio, property, init)
 # Talks to rild via the rild socket.
 unix_socket_connect(radio, rild, rild)
 
+# Talks to dbus
+unix_socket_connect(radio, dbus_daemon, dbus_daemon)
+
 # Data file accesses.
 allow radio radio_data_file:dir create_dir_perms;
 allow radio radio_data_file:notdevfile_class_set create_file_perms;
