diff --git a/rootdir/init.rc b/rootdir/init.rc
index 001ab13..659b9a4 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -561,6 +561,27 @@ service ril-daemon /system/bin/rild
     user root
     group radio cache inet misc audio sdcard_rw qcom_diag log
 
+service dbus /system/bin/dbus-daemon --system --nofork
+    disabled
+    class main
+    socket dbus stream 0660 root radio
+    user radio
+    group radio
+
+service qmiserial2qmuxd /system/bin/qmiserial2qmuxd
+    disabled
+    class main
+    user radio
+    group radio
+
+service ofonod /system/bin/ofonod --nodetach --debug
+    disabled
+    class main
+    user radio
+    group radio
+    setenv OFONO_GOBI_DEVICE /data/misc/radio/qmiserial
+    setenv OFONO_QMI_DEBUG 1
+
 service surfaceflinger /system/bin/surfaceflinger
     class core
     user system
